File: example/apple/Sources/Battery.swift
import UIKit

func printHello() {
  print("Hello from Swift!")
}

func batteryLevel() async -> Double {
  await MainActor.run {
    UIDevice.current.isBatteryMonitoringEnabled = true
    return Double(UIDevice.current.batteryLevel)
  }
}

func greet(name: String, times: Double) -> String {
  Array(repeating: "hello, \(name)", count: Int(times)).joined(separator: " ")
}

func shout(text: String) -> String {
  "\(text.uppercased())!"
}

File: example/android/src/main/kotlin/Battery.kt
import android.content.Context
import android.os.BatteryManager
import dev.flypath.FlypathApp

fun printHello() {
    println("Hello from Kotlin!")
}

suspend fun batteryLevel(): Double {
    val manager =
        FlypathApp.require().getSystemService(Context.BATTERY_SERVICE) as BatteryManager
    return manager.getIntProperty(BatteryManager.BATTERY_PROPERTY_CAPACITY) / 100.0
}

fun greet(name: String, times: Double): String =
    List(times.toInt()) { "hello, $name" }.joinToString(" ")

fun shout(text: String): String = "${text.uppercase()}!"
